query_executor.go

```diff
@@ -27,32 +27,43 @@ package gocql
 import (
 	"context"
 	"sync"
+	"sync/atomic"
 	"time"
 )
 
+// ExecutableQuery is an interface that represents a query or batch statement that
+// exposes the correct functions for the HostSelectionPolicy to operate correctly.
 type ExecutableQuery interface {
```

Comment:
What do you think about renaming type to `HostPolicyQuery`?

Reply:
I posted a comment about the renaming of this interface below in a response to James